manager/integration/tests/test_ha.py

```diff
@@ -1289,6 +1290,114 @@ def test_single_replica_restore_failure(client, core_api, volume_name, pod_make)
+
+    dr_volume_name = volume_name + "-dr"
+    client.create_volume(name=dr_volume_name, size=str(1 * Gi),
+                         numberOfReplicas=3, fromBackup=b1.url,
+                         frontend="", standby=True)
+    wait_for_volume_creation(client, dr_volume_name)
+    wait_for_volume_restoration_start(client, dr_volume_name, b1.name)
+    wait_for_volume_restoration_completed(client, dr_volume_name)
+
+    dr_volume = client.by_id_volume(dr_volume_name)
+    # Will hack into the replica directory, create a non-empty directory
+    # in the special path. (This path will be reserved for the restore.)
+    # Then the following inc restore should fail.
+    failed_replica = dr_volume.replicas[0]
+    cmd = "mkdir -p " + "/host" + failed_replica.dataPath + "/volume-delta-" +\
+          dr_volume.controllers[0].lastRestoredBackup + ".img/random-dir"
+    exec_instance_manager(core_api,
+                          failed_replica.instanceManagerName, cmd)
+
+    data_path2 = "/data/test2"
+    write_pod_volume_random_data(core_api, std_pod_name,
+                                 data_path2, DATA_SIZE_IN_MB_1)
+    std_md5sum2 = get_pod_data_md5sum(core_api, std_pod_name, data_path2)
+    snap2 = create_snapshot(client, std_volume_name)
+    std_volume.snapshotBackup(name=snap2.name)
+    wait_for_backup_completion(client, std_volume_name, snap2.name)
+    bv, b2 = find_backup(client, std_volume_name, snap2.name)
+
+    # Wait for the incremental restoration triggered then complete.
+    client.list_backupVolume()
+    check_volume_last_backup(client, dr_volume_name, b2.name)
+    wait_for_volume_restoration_start(client, dr_volume_name, b2.name)
+    wait_for_volume_restoration_completed(client, dr_volume_name)
+
+    dr_volume = client.by_id_volume(dr_volume_name)
+    verified = False
+    for r in dr_volume.replicas:
+        if r.name == failed_replica.name:
+            assert not r['running']
+            assert r['failedAt'] != ""
+            verified = True
+        else:
+            assert r['running']
+            assert r['failedAt'] == ""
+    assert verified
+    wait_for_volume_degraded(client, dr_volume_name)
+
+    activate_standby_volume(client, dr_volume_name)
+    dr_volume = wait_for_volume_detached(client, dr_volume_name)
+
+    dr_pod_name = dr_volume_name + "-pod"
+    dr_pv_name = dr_volume_name + "-pv"
+    dr_pvc_name = dr_volume_name + "-pvc"
+    dr_pod = pod_make(name=dr_pod_name)
+    create_pv_for_volume(client, core_api, dr_volume, dr_pv_name)
+    create_pvc_for_volume(client, core_api, dr_volume, dr_pvc_name)
+    dr_pod['spec']['volumes'] = [create_pvc_spec(dr_pvc_name)]
+    create_and_wait_pod(core_api, dr_pod)
+
+    md5sum1 = get_pod_data_md5sum(core_api, dr_pod_name, data_path1)
+    assert std_md5sum1 == md5sum1
+    md5sum2 = get_pod_data_md5sum(core_api, dr_pod_name, data_path2)
+    assert std_md5sum2 == md5sum2
+
+    bv.backupDelete(name=b1.name)
+    bv.backupDelete(name=b2.name)
+
```

Comment:
add a bv.delete() call to remove the backup volume. Since backup deletion and backup volume deletion are now separate actions.